### PR TITLE
resolves #45 add Darwin support

### LINKED ISSUE #45: Darwin support?
Hey @fahedouch 

Was wondering if go-logrotate could be made to work for macOS?

From a distance and without actually trying it, it looks like just providing a `logrotate_darwin` that would be the same as the _linux variant would work?

LMK if this is something you would consider merging if there was a PR.

### PR DESCRIPTION
**Pull request by slonopotamus**

### CODE REVIEW

File: logrotate_unix.go [hunk 1]
```diff
@@ -1,3 +1,5 @@
+//go:build !windows
```
Comment (fahedouch) [inline logrotate_unix.go:1]: by default it runs only for unix

File: logrotate_freebsd.go
Comment (fahedouch) [inline logrotate_freebsd.go:1]: why drop the support of freebsd ?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -25,7 +25,10 @@ jobs:
         args: --verbose
 
   test-unit:
-    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        os: [ ubuntu-20,04, macos-14 ]
```
Comment (fahedouch) [inline .github/workflows/test.yml:30]: may be adding the support of freebsd, this can be in a different PR :)

File: logrotate_freebsd.go
Comment (slonopotamus) [reply logrotate_freebsd.go:1]: It was't dropped, FreeBSD will also use `logrotate_unix.go`

File: .github/workflows/test.yml (hunk 2 above)
Comment (slonopotamus) [reply .github/workflows/test.yml:30]: Out of scope of current PR, yep.

File: logrotate_unix.go (hunk 1 above)
Comment (slonopotamus) [reply logrotate_unix.go:1]: I am not sure I understand you.

Comment (fahedouch) [reply logrotate_unix.go:1]: `go:build` tag is useless here

File: .github/workflows/test.yml (hunk 2 above)
Comment (fahedouch) [reply .github/workflows/test.yml:30]: But we still need to ensure that clown work as intended on freebsd

File: logrotate_unix.go (hunk 1 above)
Comment (slonopotamus) [reply logrotate_unix.go]: To my knowledge, `unix` is not a special filename suffix for Go.

Anyway. I reverted previous FreeBSD code because currently don't have an easy way to test that my changes work on FreeBSD.

File: .github/workflows/test.yml (hunk 2 above)
Comment (slonopotamus) [reply .github/workflows/test.yml]: I created https://github.com/fahedouch/go-logrotate/pull/52 that adds FreeBSD CI.

File: logrotate_unix.go (hunk 1 above)
Comment (fahedouch) [reply logrotate_unix.go]: the suffix `_unix.go` [was supported then dropped](https://go-review.googlesource.com/c/go/+/426296) in favor of unix build tag. 

IMO `//go:build unix` is safer

File: logrotate_unix.go [hunk 3]
```diff
@@ -1,3 +1,5 @@
+//go:build !freebsd && !windows
```
Comment (fahedouch) [inline logrotate_unix.go:1]:
```suggestion
//go:build unix
```

**Comment by slonopotamus** [comment]
Converting to draft. I'd prefer https://github.com/fahedouch/go-logrotate/pull/52 to be merged first, that will allow testing whether FreeBSD actually needs a separate codepath or we can use the same Unix code for all non-Windows platforms.

File: logrotate_unix.go (hunk 1 above)
Comment (fahedouch) [reply logrotate_unix.go]: https://github.com/fahedouch/go-logrotate/blob/main/go.mod#L3

Comment (slonopotamus) [reply logrotate_unix.go]: https://github.com/fahedouch/go-logrotate/blob/main/.github/workflows/test.yml#L33

How this works at all...

Comment (fahedouch) [reply logrotate_unix.go]: ah not good! lets make a PR to fix this before ^^

**Comment by slonopotamus** [comment]
Ready for review.

I had to bump Golang to 1.21 on CI to make `//go:build unix` work (1.21 was chosen because this is what `go.mod` currently declares). I also had to bump golangci-lint to v1.51.2 so it would properly work with Golang 1.21.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -17,23 +17,23 @@ jobs:
...
       with:
-        version: v1.45.0
+        version: v1.51.2
         args: --verbose
 
   test-unit:
     strategy:
       matrix:
-        os: [ ubuntu-20.04, windows-2022 ]
+        os: [ ubuntu-20.04, macos-14, windows-2022 ]
```
Comment (fahedouch) [inline .github/workflows/test.yml:30]:
```suggestion
        os: [ ubuntu-20.04, macOS-15, windows-2022 ]
```

[macos-15](https://github.com/actions/runner-images/blob/main/images/macos/macos-15-Readme.md) is available

**Comment by fahedouch** [comment]
LGTM, I left small comment :)

**Review by fahedouch — APPROVED** [review]
Thanks
